### ISSUE TITLE
My personal extensions are being added to an .obsolete file in C:\Users\[username]\.vscode\extensions, and no longer appear installed in VSCode.

### ISSUE THREAD

**Issue by futuremotiondev** [microsoft/vscode#224573]
Does this issue occur when all extensions are disabled?: Yes/No
Yes

- VS Code Version: 1.92.0 (user setup)
- OS Version: Windows 10 Pro x64 22H2

Steps to Reproduce:

I just today reinstalled Windows 10 and eventually installed VSCode again. Prior to reinstalling my OS, I had several personal extensions installed simply by copying the extension folders into my `.vscode\extensions` directory. They would automatically be recognized and installed despite not being packaged as VSIX files. I prefer this approach for my snippet and theme extensions as I contribute to them as I develop.

Unfortunately, this approach isn't working anymore. When I copy the extensions into the `C:\Users\[username]\.vscode\extensions` directory, they aren't recognized by VSCode. And when I restart VSCode, there is an `.obsolete` file that gets created and populated with my extension names. Essentially they are being marked as obsolete, and thus will not load.

Is this a new behavior? Just a week ago this approach was working fine. How can I get my custom extensions to install now?

Here is an example `package.json` from my theme:

```json
{
	"name": "futuremotion-light",
	"displayName": "Futuremotion Light",
	"description": "Minimal and Functional.",
	"version": "1.0.2",
	"author": "Futuremotion <admin@fmotiondev.com>",
	"publisher": "Futuremotion",
	"icon": "./img/extensionIcon.png",
	"galleryBanner": {
		"color": "#E3E6E9",
		"theme": "light"
	},
	"license": "MIT",
	"homepage": "https://github.com/futuremotiondev/fm-vscode-futuremotion-light",
	"repository": {
		"type": "git",
		"url": "https://github.com/futuremotiondev/fm-vscode-futuremotion-light.git",
		"docs": "https://github.com/futuremotiondev/fm-vscode-futuremotion-light/blob/main/README.md"
	},
	"bugs": {
		"url": "https://github.com/futuremotiondev/fm-vscode-futuremotion-light/issues"
	},
	"engines": {
		"vscode": "^1.86.0"
	},
	"categories": [
		"Themes"
	],
	"keywords": [
		"theme",
		"Light Theme",
		"Futuremotion"
	],
	"contributes": {
		"themes": [
			{
				"label": "Futuremotion Light",
				"uiTheme": "vs",
				"path": "./themes/futuremotion-light.json"
			}
		]
	},
	"devDependencies": {
		"@changesets/changelog-github": "^0.5.0",
		"@changesets/cli": "^2.27.1",
		"@primer/primitives": "^7.15.12",
		"@vscode/vsce": "^2.24.0",
		"chroma-js": "^2.4.2",
		"color": "^4.2.3",
		"nodemon": "^3.1.0"
	},
	"scripts": {
		"package": "mkdir build && vsce package -o ./build/fm-theme-futuremotion-light.vsix",
		"prepublishOnly": "vsce publish"
	},
	"__metadata": {
		"installedTimestamp": 1711774234540
	}
}
```

`.obsolete` file:

```
{"futuremotion.futuremotion-light-1.0.2":true}
```

How can I get around my extension being marked as obsolete? Do I *really* need to package as a VSIX now?

Any help would be greatly appreciated.

**Comment by futuremotiondev** [comment]
Ok, an update - I managed to fix it. 

For anyone else running into this issue, this is what I did: 

1. Disabled auto-update for all extensions.
2. Replaced the fresh `extensions.json` file with my old `extensions.json` file from my previous install.

I think disabling the default "auto-update extensions" behavior did it, but I can't be sure.

Now when I drop non-VSIX plugin folders into `.vscode\extensions` they correctly appear as installed.
